src/compiler/ast/statement/Statement.ts

```diff
@@ -9,4 +10,27 @@ export class Statement<T extends ts.Statement = ts.Statement> extends Node<T> {
     remove() {
         removeStatementedNodeChild(this);
     }
+
+    /**
+     * Will move the statement to a new file updating all the references in the project.
+     *
+     * If `nodeFilePath` is not provided then the name of the new file will be the one of the first symbol
+     * found or `newFile` if none is found.
+     *
+     * WARNING! This will forget all the nodes in this node's file! It's best to do this after you're all done with the file.
+     *
+     * @returns the new SourceFile to which the statement was moved or undefined if the statement coulnd't be moved.
+     */
+    moveToNewFile(newFilePath?: string): SourceFile | undefined {
```

Comment:
Could you remove this method from the PR? I don't want to suggest these refactors by providing an easy way to do them on the objects because they cause nodes to be forgotten.

Reply:
Perhaps we could add them some other way in the future, but I don't think it should be so readily available directly on `Statement`